source/Coop/Mod/CoopServer.cs

```diff
@@ -265,25 +279,61 @@ private void ReceiveClientRequestWorldData(ConnectionBase connection, Packet pac
                     connectionServer);
         }
 
-        [GameServerPacketHandler(ECoopServerState.Preparing, EPacket.Client_DeclineWorldData)]
-        private void ReceiveClientDeclineWorldData(ConnectionBase connection, Packet packet)
+        [GameServerPacketHandler(ECoopServerState.Preparing, EPacket.Client_RequestGameData)]
+        private void SendGameData(ConnectionBase connection, Packet packet)
         {
-            m_CoopServerSMs[(ConnectionServer)connection].StateMachine.Fire(ECoopServerTrigger.DeclineWorldData);
+            PlayerHeroSerializer serializer = (PlayerHeroSerializer)CommonSerializer.Deserialize(packet.Payload);
+            Hero newHero = (Hero)serializer.Deserialize();
+            Guid newHeroGuid = CoopObjectManager.AddObject(newHero);
+
+            CoopObjectManager.AddObject(newHero);
```

Comment:
Since I'm planning on reworking this I'm not too worried about it, but this seems like it would add dupes once we get sync going more, esp around homesettlement and such.

Reply:
AddObject manages that. It is impossible to create duplicates